payments/models.py

```diff
@@ -105,46 +81,95 @@ class Meta:
+    min_width = models.DecimalField(
+        verbose_name=_("minimum width"), max_digits=5, decimal_places=2,
     )
-    harbor = models.ForeignKey(
-        "resources.Harbor",
-        verbose_name=_("harbor"),
-        related_name="products",
-        on_delete=models.CASCADE,
-        null=True,
-        blank=True,
+    max_width = models.DecimalField(
+        verbose_name=_("maximum width"), max_digits=5, decimal_places=2,
+    )
+    tier_1_price = models.DecimalField(
+        verbose_name=_("tier 1 price"),
+        max_digits=10,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.00"))],
+    )
+    tier_2_price = models.DecimalField(
+        verbose_name=_("tier 2 price"),
+        max_digits=10,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.00"))],
     )
+    tier_3_price = models.DecimalField(
+        verbose_name=_("tier 3 price"),
+        max_digits=10,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.00"))],
+    )
+    price_unit = models.CharField(
+        choices=[(PriceUnits.AMOUNT, PriceUnits.AMOUNT.label)],
+        default=PriceUnits.AMOUNT,
+        max_length=10,
+    )
+    objects = BerthProductManager()
 
     class Meta:
         constraints = [
             UniqueConstraint(
-                fields=["price_group", "harbor"],
-                name="unique_product_for_harbor_pricegroup",
-            ),
+                fields=("min_width", "max_width",), name="unique_width_range"
+            )
         ]
+        ordering = ["min_width"]
 
     def __str__(self):
-        harbor_name = str(self.harbor) + " " if self.harbor else ""
-        return f"{self.price_group.name} - {harbor_name}({self.price_value}€)"
+        return (
+            f"[{self.min_width}-{self.max_width}] "
+            f"T1:{self.tier_1_price}€, T2:{self.tier_2_price}€, T3: {self.tier_3_price}€"
+        )
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        super().save(*args, **kwargs)
+
+    @property
+    def display_dimensions(self):
+        return f"{self.min_width}m - {self.max_width}m"
 
     @property
     def name(self):
         return _("Berth product") + f": {self}"
 
+    def price_for_tier(self, tier: PriceTier) -> Decimal:
+        if tier == PriceTier.TIER_1:
+            return self.tier_1_price
+        elif tier == PriceTier.TIER_2:
+            return self.tier_2_price
+        elif tier == PriceTier.TIER_3:
+            return self.tier_3_price
+        raise ValueError(_("Tier not implemented"))
+
 
-class WinterStorageProduct(AbstractPlaceProduct):
+class WinterStorageProduct(AbstractPlaceProduct, AbstractBaseProduct):
     winter_storage_area = models.OneToOneField(
         "resources.WinterStorageArea",
         verbose_name=_("winter storage area"),
         related_name="product",
         on_delete=models.CASCADE,
     )
+    price_unit = models.CharField(
```

Comment:
Should `price_unit` to one of the abstact base classes? This field is duplicated in `WinterStorageProduct` and `BerthProduct`.

Reply:
I tried adding it to the `AbstractPlaceProduct` class but it has some conflict, I guess because also `AbstractBaseProduct` implements it and it doesn't know which one has precedence.